### ISSUE TITLE
Unexpected values returned by MeanMetricWrapper subclass

### ISSUE THREAD

**Issue by AakashKumarNain** [keras-team/keras-core#32]
I have a very interesting test case that I am unable to debug. I implemented the `Hinge` loss (check #31 ), and imported the `hinge(...)` function for the corresponding metric implementation. The values that are coming out of metric are way off than the actual values but the values coming out of the loss function are as expected. Here is the test case:

```python
y_true = tf.constant([[-1., 1., -1., 1.], [-1., -1., 1., 1.]])
y_pred = tf.constant([[-0.3, 0.2, -0.1, 1.6], [-0.25, -1.0, 0.5, 0.6]])
sample_weight = tf.constant([1.5, 2.0])

##### Hinge loss #########################

# TF implementation
tf.keras.losses.Hinge()(y_true, y_pred, sample_weight)
# output <tf.Tensor: shape=(), dtype=float32, numpy=0.8625>

# My implemetation
keras_core.losses.Hinge()(y_true, y_pred, sample_weight)
# ouput <tf.Tensor: shape=(), dtype=float32, numpy=0.8625>

##### Hinge metric #########################

# TF metric implementation
tf.keras.metrics.Hinge()(y_true, y_pred, sample_weight)
# output <tf.Tensor: shape=(), dtype=float32, numpy=0.49285716> 

# My metric implementation
keras_core.metrics.Hinge()(y_true, y_pred, sample_weight=sample_weight)
# output <tf.Tensor: shape=(), dtype=float32, numpy=0.575>
```

Here is how the the class implementation looks like:

```python
@keras_core_export("keras_core.metrics.Hinge")
class Hinge(reduction_metrics.MeanMetricWrapper):
    def __init__(self, name="hinge", dtype=None):
        super().__init__(fn=keras_core.losses.hinge, name=name, dtype=dtype)

    def get_config(self):
        return {"name": self.name, "dtype": self.dtype}
```

**Expected behavior:**

Given that the loss function is working as expected, the metric implementation should work as we are just importing the same function in the corresponding metric class
